### PR TITLE
Issue 477 skip py27

### LINKED ISSUE #477: Shall we skip Python 2.7 support?
# Description

Supporting only Python 3.x makes the code easier to maintain. In `OWSLib` we have already done this step.

What is your opinion?

### PR DESCRIPTION
**Pull request by cehbrecht**
# Overview

This PR skips Python 2.7 support.

Changes:
* removed `_compat` module.
* removed usage of `six`.
* run 2to3.

# Related Issue / Discussion

Fix #477 and #509.

# Additional Information

# Contribution Agreement

(as per https://github.com/geopython/pywps/blob/master/CONTRIBUTING.rst#contributions-and-licensing)

- [ ] I'd like to contribute [feature X|bugfix Y|docs|something else] to PyWPS. I confirm that my contributions to PyWPS will be compatible with the PyWPS license guidelines at the time of contribution.
- [x] I have already previously agreed to the PyWPS Contributions and Licensing Guidelines

### CODE REVIEW

**Comment by cehbrecht** [comment]
@davidcaron @Zeitsperre Do you like to have a look at this PR?

File: tests/test_execute.py [hunk 1]
```diff
@@ -295,7 +290,7 @@ def test_input_parser(self):
         """
         my_process = create_complex_proces()
         service = Service(processes=[my_process])
-        self.assertEqual(len(service.processes.keys()), 1)
+        self.assertEqual(len(list(service.processes.keys())), 1)
```
Comment (davidcaron) [inline tests/test_execute.py]: could be replaced with `len(service.processes)`

File: tests/test_execute.py [hunk 2]
```diff
@@ -352,7 +347,7 @@ def test_input_default(self):
...
```
Comment (davidcaron) [inline tests/test_execute.py]: could be replaced with `len(service.processes)`

File: tests/test_execute.py [hunk 3]
```diff
@@ -376,7 +371,7 @@ def test_output_mimetype(self):
         """
         my_process = create_mimetype_process()
         service = Service(processes=[my_process])
-        self.assertEqual(len(service.processes.keys()), 1)
+        self.assertEqual(len(list(service.processes.keys())), 1)
```
Comment (davidcaron) [inline tests/test_execute.py]: could be replaced with `len(service.processes)`

File: pywps/translations.py [hunk 4]
```diff
@@ -44,4 +44,4 @@ def lower_case_dict(translations=None):
     """
     if translations is None:
         return
-    return {k.lower(): v for k, v in translations.items()}
+    return {k.lower(): v for k, v in list(translations.items())}
```
[the remark this answers is not in the document]
Comment (cehbrecht) [reply pywps/translations.py]: done.

File: pywps/response/capabilities.py [hunk 5]
```diff
@@ -20,7 +20,7 @@ def json(self):
         """Convert the response to JSON structure
         """
 
-        processes = [p.json for p in self.processes.values()]
+        processes = [p.json for p in list(self.processes.values())]
```
[the remark this answers is not in the document]
Comment (cehbrecht) [reply pywps/response/capabilities.py]: done.

File: tests/test_execute.py (hunk 1 above)
Comment (cehbrecht) [reply tests/test_execute.py]: done.

File: tests/test_execute.py (hunk 2 above)
Comment (cehbrecht) [reply tests/test_execute.py]: done.

File: tests/test_execute.py (hunk 3 above)
Comment (cehbrecht) [reply tests/test_execute.py]: done.

File: pywps/wpsserver.py [hunk 6]
```diff
@@ -23,12 +23,10 @@ def temp_dir():
         shutil.rmtree(tmp)
 
 
-class PyWPSServerAbstract(object):
+class PyWPSServerAbstract(object, metaclass=ABCMeta):
```
[the remark this answers is not in the document]
Comment (cehbrecht) [reply pywps/wpsserver.py]: Yes. Looks like old code ... also not used in Flask demo. I would remove it in another PR.

**Comment by cehbrecht** [comment]
@davidcaron @Zeitsperre @tomkralidis thanks for review :)

**Pull request merged**